manager/orchestrator/service.go

```diff
@@ -44,8 +46,23 @@ func DeleteServiceTasks(ctx context.Context, s *store.MemoryStore, service *api.
 	err = s.Batch(func(batch *store.Batch) error {
 		for _, t := range tasks {
 			err := batch.Update(func(tx store.Tx) error {
-				if err := store.DeleteTask(tx, t.ID); err != nil {
-					log.G(ctx).WithError(err).Errorf("failed to delete task")
+				// time travel is not allowed. if the current desired state is
```

Comment:
A bug should be filed to have a function that enforces this "math".

Reply:
Albeit, "desired state" doesn't have the same constraints as the state field, so this is less important.

"DesiredState" is typically directly observable.

Reply:
I'm unclear what you mean by this.

Reply:
Sorry for the lack of clarity. I was just pointing out that we should probably always do this check when we change the state. I am not sure it should be done in this PR, as it may introduce a regression.